mscore/file.cpp

```diff
@@ -2540,76 +2540,80 @@ bool MuseScore::saveSvg(Score* score, const QString& saveName)
-                }
-                else { // Draw staff lines once per system
-                    StaffLines* firstSL = s->firstMeasure()->staffLines(i)->clone();
-                    StaffLines*  lastSL =  s->lastMeasure()->staffLines(i);
-                    firstSL->bbox().setRight(lastSL->bbox().right()
-                                          +  lastSL->pagePos().x()
-                                          - firstSL->pagePos().x());
-                    printer.setElement(firstSL);
-                    paintElement(p, firstSL);
-                }
-            }
-        }
-        // 2nd pass: the rest of the elements
-        QList<const Element*> pel = page->elements();
-        qStableSort(pel.begin(), pel.end(), elementLessThan);
-
-        Element::Type eType;
-        foreach (const Element* e, pel) {
-            // Always exclude invisible elements
-            if (!e->visible())
-                    continue;
-
-            eType = e->type();
-            switch (eType) { // In future sub-type code, this switch() grows, and eType gets used
-            case Element::Type::STAFF_LINES : // Handled in the 1st pass above
-                continue; // Exclude from 2nd pass
-                break;
-            default:
-                break;
-            } // switch(eType)
+                        if (byMeasure) { // Draw visible staff lines by measure
+                              for (MeasureBase* mb = s->firstMeasure(); mb != 0; mb = s->nextMeasure(mb)) {
+                                    if (mb->type() != Element::Type::HBOX
+                                     && mb->type() != Element::Type::VBOX
+                                     && static_cast<Measure*>(mb)->visible(i)) {
+                                          StaffLines* sl = static_cast<Measure*>(mb)->staffLines(i);
+                                          printer.setElement(sl);
+                                          paintElement(p, sl);
+                                    }
+                              }
+                        }
+                        else { // Draw staff lines once per system
+                              StaffLines* firstSL = s->firstMeasure()->staffLines(i)->clone();
+                              StaffLines*  lastSL =  s->lastMeasure()->staffLines(i);
+                              firstSL->bbox().setRight(lastSL->bbox().right()
+                                                    +  lastSL->pagePos().x()
+                                                    - firstSL->pagePos().x());
+                              printer.setElement(firstSL);
+                              paintElement(p, firstSL);
+                        }
+                  }
+            }
+            // 2nd pass: the rest of the elements
+            QList<const Element*> pel = page->elements();
+            qStableSort(pel.begin(), pel.end(), elementLessThan);
+
+            Element::Type eType;
+            foreach (const Element* e, pel) {
```

Comment:
while touching the code, you could as well change the all the `foreach (a, b)` into `for (a : b)`